packages/11ty/_plugins/figures/index.js

```diff
@@ -10,8 +10,8 @@ module.exports = function (eleventyConfig, options = {}) {
     let { figure_list: figureList } = eleventyConfig.globalData.figures
     figureList = await Promise.all(
       figureList.map((data) => {
-        const figure = new Figure(eleventyConfig, data)
-        return figure.create()
+        const factory = new FigureFactory(eleventyConfig, data)
+        return factory.create()
       })
```

Comment:
Instead of creating a `factory` for every figure the class constructor can be used to configure the factory with the `eleventyConfig` and then pass `data` for each figure individually to the `create` method.